var/spack/repos/builtin/packages/samrai/package.py

```diff
@@ -53,24 +55,26 @@ class Samrai(Package):
     depends_on("zlib")
     depends_on("hdf5+mpi")
     depends_on("boost")
+    depends_on("m4")
```

Comment:
It might not be if there is a `--with-M4=prefix` flag. But I don't know enough about the code.